include/deal.II/base/parallel.h

```diff
@@ -549,71 +549,108 @@ namespace parallel
                           const Function                             &f,
                           const unsigned int                          grainsize)
   {
-#ifdef DEAL_II_WITH_TASKFLOW
-    tf::Executor &executor = MultithreadInfo::get_taskflow_executor();
-    tf::Taskflow  taskflow;
+    Assert(grainsize > 0, ExcMessage("Grainsize must be a positive number."));
+
+    bool worth_it = (MultithreadInfo::n_threads() > 1);
 
     if constexpr (std::is_integral_v<Iterator>)
       {
-        // Integral "iterator" types (e.g., int): use plain arithmetic
-        using integral_type = Iterator;
-
-        const integral_type           n = end - begin;
-        tf::IndexRange<integral_type> range(0, n, 1);
-
-        taskflow.for_each_by_index(
-          range,
-          [&, begin](const tf::IndexRange<integral_type> &subrange) {
-            integral_type subrange_begin =
-              begin + static_cast<integral_type>(subrange.begin());
-
-            integral_type subrange_end =
-              subrange_begin + static_cast<integral_type>(subrange.size());
-
-            f(subrange_begin, subrange_end);
-          },
-          tf::GuidedPartitioner(grainsize));
+        if ((end - begin) < static_cast<Iterator>(grainsize))
+          worth_it = false;
       }
     else
       {
-        // Non-integral iterator types: advance from 'begin' using
-        // std::distance/std::advance
-        using diff_t = typename std::iterator_traits<Iterator>::difference_type;
+        if (std::distance(begin, end) < grainsize)
```

Comment:
I wonder whether `std::distance` is expensive enough that it's worth creating a variable `n_applications` or some such at the top that is either `(end-begin)` or `std::distance(begin,end)`, and then not have to call it ever again below.

Reply:
Hopefully not, but consider it done.